woocommerce/woocommerce
# Adapt receipt to take into account order status

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR adds a new `Payment status` field to the order receipts. Paid orders will show as the screenshot in the right column 👇 (notice the `paid` in the diff section titles).

| Failed order | Paid order |
|--------|--------|
|<img width="628" alt="Screenshot 2024-10-21 at 11 17 38" src="https://github.com/user-attachments/assets/81cafc81-13d3-408b-8cad-1b27a7ba4fca">| <img width="661" alt="Screenshot 2024-10-21 at 11 15 08" src="https://github.com/user-attachments/assets/497eea5b-56c6-4038-b259-b307d9f54962">| 

Closes https://github.com/woocommerce/woocommerce/issues/51813



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install the `WooCommerce Stripe Gateway` plugin.
2. Check the `Enable Stripe` and `Enable test mode` options in the settings.
3. Go to the store, add a product to the cart, and go to the checkout.
4. Use a [decline test card](https://docs.stripe.com/testing#declined-payments) as a payment method and try to place the order. You should see an error indicating the order failed.
5. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys` and add a new key with `Read/Write` permissions.
6. Make the following request using the consumer key and secret generated in the previous step:
```bash
curl --insecure -u [KEY]:[SECRET] --request POST --url 'https://store.local/wp-json/wc/v3/orders/33/receipt?force_new=true' --header 'Content-Type: application/json'
```
7. It should respond with something similar to:
```json
{
  "receipt_url": "https://store.local/wc/file/transient/XXXXXXXXX",
  "expiration_date": "2024-10-15"
}
```
8. Copy the `receipt_url` and open it in a browser.
9. Check that it looks like the `Failed order` image in the PR description. It should clearly show that the receipt is for an order with `Failed` status.
10. Repeat the process using a valid test card so the order doesn't fail and check the receipt looks light the `Paid receipt` one above.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Adapt receipt to take into account failed orders.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
